streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -641,6 +678,19 @@ public Serde valueSerde() {
         return parsed;
     }
 
+    @Override
+    public Long getLong(final String key) {
+        // return different default value for `commit.interval.ms` if EOS enabled
+        if (StreamsConfig.COMMIT_INTERVAL_MS_CONFIG.equals(key)
+            && eosEnabled
+            // use EOS default value only if no user config is set
+            && !originals().containsKey(StreamsConfig.COMMIT_INTERVAL_MS_CONFIG)) {
+            return EOS_DEFAULT_COMMIT_INTERVAL_MS;
```

Comment:
Also, if the config has the right value, it will be part of what gets logged by default and it's then less important to log the fact that we are picking a different default. Still, these things make it harder for support people so it's not a clear call. Maybe a debug log would still be useful.

You refer to the fact that this is documented, but every support person knows that a significant number of people don't read the docs.